kolibri/core/content/models.py

```diff
@@ -140,6 +142,38 @@ def __str__(self):
         return self.tag_name
 
 
+class ContentNodeQueryset(TreeQuerySet):
+
+    def dedupe_by_content_id(self):
+        # remove duplicate content nodes based on content_id
+        if OPTIONS['Database']["DATABASE_ENGINE"] == "sqlite":
+            # filter by ids for the deduplicated content nodes
+            deduped_list = []
+            content_ids = set()
+            for node in self:
+                if node.content_id not in content_ids:
+                    deduped_list.append(node)
+                    content_ids.add(node.content_id)
+            return self.filter(id__in=[node.id for node in deduped_list])
+
+        # when using postgres, we can call distinct on a specific column
+        elif OPTIONS['Database']["DATABASE_ENGINE"] == "postgres":
+            return self.order_by('content_id').distinct('content_id')
+
+
+class ContentNodeManager(models.Manager.from_queryset(ContentNodeQueryset), TreeManager):
+
+    def get_queryset(self, *args, **kwargs):
```

Comment:
taken from https://github.com/django-mptt/django-mptt/blob/master/mptt/managers.py#L93
to let users know that this is the standard implementation for the `get_queryset` method on the tree manager.